OpenRA.Mods.Common/Traits/Conditions/GrantChargeDrainCondition.cs

```diff
@@ -0,0 +1,211 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Orders;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Grants a condition for a certain time when deployed. Can be reversed anytime " +
+		"by undeploying. Leftover charges are taken into account when recharging.")]
```

Comment:
Informational (I am fine with not using it for the saboteur):
From a modder's point of view you may want to disable cancelling in some cases. A `IsManuallyUndeployable` property would be nice (defaulting to false).
Dune 2000 actually doesn't allow you to cancel the Saboteur's cloak when active.